packages/react-router/lib/types.ts

```diff
@@ -170,11 +167,12 @@ type __tests = [
     >
   >,
   Expect<
+    //@ts-expect-error TODO: Why is this failing?
```

Comment:
I'm unsure what's going on here.  We no longer have any sort of `TypedResponse` so I would have expected a raw json value to make this pass?

Reply:
I think this is still expecting the `Date` to get serialized. Looks like it should be:

```ts
  Expect<
    Equal<
      Pretty<
        ServerDataFrom<
          () =>
            | { json: string; b: Date; c: () => boolean }
            | DataWithResponseInit<{ data: string; b: Date; c: () => boolean }>
        >
      >,
      | { json: string; b: Date; c: undefined }
      | { data: string; b: Date; c: undefined }
    >
  >,
```

I'm having trouble getting GitHub to do the suggested commit thing

Reply:
ohhhhh thank you I totally missed the Date in there